oxidation/libparsec/crates/types/src/error.rs

```diff
@@ -2,9 +2,21 @@
 
 use thiserror::Error;
 
-use crate::{DateTime, DeviceID, EntryID, RealmID, UserID};
+use crate::{DateTime, DeviceID, EntryID, RealmID, Regex, UserID};
 use libparsec_crypto::CryptoError;
 
+#[derive(Error, Debug)]
+pub enum RegexError {
+    #[error("Regex parsing err: {err}")]
+    ParseError { err: regex::Error },
+    #[error("Failed to convert glob pattern into regex: {err}")]
+    GlobPatternError { err: fnmatch_regex::error::Error },
+    #[error("IO error while opening pattern file: {err}")]
+    PatternFileIOError { err: std::io::Error },
```

Comment:
```suggestion
    #[error("IO error on pattern file `{file}`: {err}")]
    PatternFileIOError { file: std::path::PathBuf, err: std::io::Error },
```

:thinking:

Reply:
I'll let you judge on that proposed change

Reply:
It makes the error message better to have the file path.